autotest/ogr/ogr_mem.py

```diff
@@ -709,6 +709,49 @@ def test_ogr_mem_alter_geom_field_defn():
     assert lyr.GetSpatialRef() is None
 
 
+###############################################################################
+# Test ogr.Layer.__arrow_c_stream__() interface.
+# Cf https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html
+
+
+@gdaltest.enable_exceptions()
+def test_ogr_mem_arrow_stream_pycapsule_interface():
+
+    ds = ogr.GetDriverByName("Memory").CreateDataSource("")
+    lyr = ds.CreateLayer("foo")
+
+    stream = lyr.__arrow_c_stream__()
+    assert stream
+    t = type(stream)
+    assert t.__module__ == "builtins"
+    assert t.__name__ == "PyCapsule"
```

Comment:
This object is indeed not accessible from Python, and so you have to do something like the above to verify it. But the `ctypes` module does provide some interface to C API functions, and you can do:

```
ctypes.pythonapi.PyCapsule_IsValid(ctypes.py_object(capsule), name)
```

Now, it doesn't add that much of value if you also test that it is consumed correctly by pyarrow, which ensures that it's a valid capsule with the correct name as well.

Reply:
thanks. I've added a variation of the above ctypes